airbyte-cdk/python/airbyte_cdk/sources/declarative/yaml_declarative_source.py

```diff
@@ -69,6 +70,27 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
             self._apply_log_level_to_stream_logger(self.logger, stream)
         return source_streams
 
+    def spec(self, logger: logging.Logger) -> ConnectorSpecification:
+        """
+        Returns the spec for this integration. The spec is a JSON-Schema object describing the required configurations (e.g: username
+        and password) required to run this integration. For low-code connectors, this will first attempt to load the spec from the
+        manifest's spec block, otherwise it will load it from "spec.yaml" or "spec.json" in the project root.
+        """
+
+        self.logger.debug(
+            "parsed YAML into declarative source",
+            extra={"path_to_yaml_file": self._path_to_yaml, "source_name": self.name, "parsed_config": json.dumps(self._source_config)},
+        )
+
+        spec = self._source_config.get("spec")
+        if spec:
+            if "class_name" not in spec:
+                spec["class_name"] = "airbyte_cdk.sources.declarative.spec.Spec"
```

Comment:
what if the developer uses a `type` field instead of `class_name`?

Reply:
All three formats should work. If they define the `class_name` then we'll just use that. We define `Spec` in the `class_types_registry` so it should work with `type`, and if it's omitted here then we'll insert class name here.